src/types/account-manager.ts

```diff
@@ -438,7 +438,18 @@ export class AccountManager {
       return nacl.sign.detached(bytesToSign, keypair.secretKey)
     }
 
-    return this.signerAccount(generateAddressWithSigners({ ed25519Pubkey: keypair.publicKey, rawEd25519Signer: rawSigner }))
+    const addrWithSigners = generateAddressWithSigners({ ed25519Pubkey: keypair.publicKey, rawEd25519Signer: rawSigner })
+
+    this._accounts[addrWithSigners.addr.toString()] = addrWithSigners
+
+    const acct = addrWithSigners.addr as Address & AddressWithSigners
+
+    for (const prop in addrWithSigners) {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      ;(acct as any)[prop] = (addrWithSigners as any)[prop]
```

Comment:
This is pretty gross but it's needed for backwards compatibility. Tbh I don't think we should have ever injected properties into an existing class, so maybe this could be a candidate for a breaking change.